src/main/java/domain/Invoice.kt

```diff
@@ -5,24 +5,23 @@ import domain.enums.InvoiceState
 
 import javax.persistence.*
 import java.io.Serializable
-import java.util.Date
+import java.util.*
 
 @Entity
 @NamedQuery(name = "Invoice.allInvoices", query = "SELECT i FROM Invoice i")
-@Table(name = "Invoice")
+@Table(name = "INVOICE")
 class Invoice : Serializable {
 
     @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
-    var id: Long? = null
+    var id: String? = UUID.randomUUID().toString()
     @Column
     var createdOn: Date? = null
     @Column
     var generatedFor: Date? = null
     @Enumerated(EnumType.STRING)
     var generationType: InvoiceGenerationType? = null
     @Enumerated(EnumType.STRING)
-    var state: InvoiceState? = null
+    var state: InvoiceState? = InvoiceState.OPEN
```

Comment:
If the invoice status defaults to OPEN, it should not be nullable

Reply:
DONE